src/pybind/mgr/dashboard/frontend/src/app/shared/api/rbd.service.ts

```diff
@@ -40,23 +43,30 @@ export class RbdService {
     return this.http.get(`api/block/image/${imageSpec.toStringEncoded()}`);
   }
 
-  list() {
-    return this.http.get<RbdPool[]>('api/block/image').pipe(
-      map((pools) =>
-        pools.map((pool) => {
-          pool.value.map((image) => {
-            if (!image.configuration) {
+  list(params: any) {
+    return this.http
+      .get<RbdPool[]>('api/block/image', {
+        params: params,
+        headers: { Accept: this.getVersionHeaderValue(2, 0) },
+        observe: 'response'
+      })
+      .pipe(
+        map((response: any) => {
+          return response['body'].map((pool: any) => {
+            pool.value.map((image: any) => {
+              if (!image.configuration) {
+                return image;
+              }
+              image.configuration.map((option: any) =>
+                Object.assign(option, this.rbdConfigurationService.getOptionByName(option.name))
+              );
               return image;
-            }
-            image.configuration.map((option) =>
-              Object.assign(option, this.rbdConfigurationService.getOptionByName(option.name))
-            );
-            return image;
+            });
+            pool['headers'] = response.headers;
+            return pool;
```

Comment:
Given we're modifying the existing rbd list API call in a backward incompatible way, what about cleaning up the response we're generating at the backend and providing something ready to be displayed in the front-end?

Personally, except for things like debouncing or similar, I think we shouldn't do any processing of the back-end data in the front-end (just cosmetic stuff, like pipes, widgets, hiding columns, and so). Back-end data should fit 1-1 with front-end layout.

Reply:
https://tracker.ceph.com/issues/56557